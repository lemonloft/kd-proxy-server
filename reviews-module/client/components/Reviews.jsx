/* eslint-disable */
import React from 'react';
import styled from 'styled-components';
import HostResponse from './HostResponse.jsx';
import PageCarousel from './PageCarousel.jsx';

const Avatar = styled.img`
border-radius: 50%;
`;

const Body = styled.div`
display: flex;
flex-direction: column;
`;

const Review = styled.div`
margin-bottom: 24px;
`;

const ReviewText = styled.div`
width: 594px;
`

const HorizontalTest = styled.div`
display: flex;
flex-direction: row;
`;

const VerticalTest = styled.div`
display: flex;
flex-direction: column;
margin-left: 15px;

justify-content: center;
`;

const ReviewButton = styled.button`
outline:none;
border:none;
color: #378187;
font-size: 16px;
padding: 0;
font-family: Circular, -apple-system, BlinkMacSystemFont, Roboto, Helvetica Neue, sans-serif;
:hover {
    text-decoration: underline;
}
`;

const HorLine = styled.div`
  width: 594px;
  border-bottom: 1px solid #BEBEBE;
  margin-top: 25px;
`;

const Margin = styled.div`
margin-top: 10px;
`;

class Reviews extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
        isClicked: this.props.data.map((data) => {return false}),
        revPageNum: 1,
        serPageNum: 1,
    };
  }

  resetSerPageNum() {
    this.setState({
        serPageNum: 1,
    });
  }

  getProperDate(dateString) {
    const monthNames = [
      '', 'January', 'February', 'March',
      'April', 'May', 'June', 'July',
      'August', 'September', 'October',
      'November', 'December',
    ];
    const date = new Date(dateString);
    return `${monthNames[date.getMonth()] } ${date.getFullYear()}`;
  }

  changeReadmoreStatus(e, index) {
    let oldIsClicked = this.state.isClicked.slice();
    let newIsClicked = oldIsClicked.map((data, i) => {
        if (i === index) {
            return true;
        }
        return data;
    })
    this.setState({
        isClicked: newIsClicked,
    });
  }

  changePageNum(pageNum) {
    if (!this.props.searchBool) {
        window.scroll({top: 400, left: 0, behavior: 'smooth' });
        this.setState({
          revPageNum: pageNum,
        })
    } else if (this.props.searchBool) {
        window.scroll({top: 100, left: 0, behavior: 'smooth' });
        this.setState({
          serPageNum: pageNum,
        })    
    }
  }

  leftPageNum() {
    if (!this.props.searchBool) {
        window.scroll({top: 400, left: 0, behavior: 'smooth' });
        let rPN = this.state.revPageNum;
        this.setState({
          revPageNum: rPN - 1,
        })
    } else if (this.props.searchBool) {
        window.scroll({top: 100, left: 0, behavior: 'smooth' });
        let sPN = this.state.serPageNum;
        this.setState({
          serPageNum: sPN - 1,
        })   
    }
  }

  rightPageNum() {
    if (!this.props.searchBool) {
        window.scroll({top: 400, left: 0, behavior: 'smooth' });
        let rPN = this.state.revPageNum;
        this.setState({
          revPageNum: rPN + 1,
        })
    } else if (this.props.searchBool) {
        window.scroll({top: 100, left: 0, behavior: 'smooth' });
        let sPN = this.state.serPageNum;
        this.setState({
          serPageNum: sPN + 1,
        })   
    }
  }

  render() {
    let pageNum = this.props.searchBool ? this.state.serPageNum : this.state.revPageNum;
    let pageData = this.props.data.slice((pageNum-1)*7, pageNum*7);
    return (
      <Body>
        {pageData.map((data, index) => (
          <Review key={data._id}>
            <HorizontalTest>
                <a href="/">
                    <Avatar src={data.image} height='40px' width='40px' />
                </a>
                <VerticalTest>
                  <div><b>{data.name}</b></div>
                  <span>{this.getProperDate(data.date)}</span>
                </VerticalTest>
            </HorizontalTest>
            <Margin />
            <ReviewText>
                <span>
                {data.body.length <= 275 &&
                    data.body
                }
                {data.body.length > 275 &&
                (this.state.isClicked[index] ? data.body.slice(0,275) + data.body.slice(275).split(' ')[0] : data.body.slice(0,275) + data.body.slice(275).split(' ')[0] + '...')
                }
                {data.body.length > 275 &&
                    (this.state.isClicked[index] ? (' ' + data.body.slice(275).split(' ').slice(1).join(' ')) : (<ReviewButton onClick={(e)=>{this.changeReadmoreStatus(e, index)}}>Read more</ReviewButton>))
                }
                <HostResponse data={data} />
                <HorLine />
                </span>
            </ReviewText>
          </Review>
        ))}
        <PageCarousel searchBool={this.props.searchBool} changePageNum={this.changePageNum.bind(this)} serPageNum={this.state.serPageNum} revPageNum={this.state.revPageNum} end={Math.ceil(this.props.data.length/7)} leftPageNum={this.leftPageNum.bind(this)} rightPageNum={this.rightPageNum.bind(this)} />
      </Body>
    );
  }
}

export default Reviews;
