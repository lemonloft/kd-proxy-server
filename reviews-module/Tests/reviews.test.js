import React from 'react';
import App from '../client/app.jsx';
import Reviews from '../client/components/Reviews.jsx';
import { shallow, mount } from 'enzyme';

describe('it App component using shallow', () => {
  let shallowWrapper;
  beforeAll(() => {
    shallowWrapper = shallow(<App />);
    console.log(shallowWrapper.debug());
  });

  it('should not return an error', () => {
    expect(shallowWrapper).toMatchSnapshot();
  });
});

describe('it App component using mount', () => {
  let mountWrapper;
  beforeAll(() => {
    mountWrapper = mount(<App />);
    console.log(mountWrapper.debug());
  });
  it('should not return an error', () => {
    expect(mountWrapper).toMatchSnapshot();
  });
  it('should have an "APP" component "at" index of 0', () => {
    expect(mountWrapper.at(0).type()).toBe(App);
  });
});

let state = {
  data: [{
    accuracy: 4,
    amaAme: 1,
    body: 'Doloribus in illum eos consectetur et. Possimus ut cumque est quae sit. Error dolorem laboriosam beatae. Vel et sequi vel sint rerum commodi omnis optio sit. Ut eius molestias quibusdam temporibus sit rem qui. Omnis dolorem enim.',
    checkin: 5,
    cleanliness: 3,
    communication: 4,
    date: '2019-11-30T11:29:24.000Z',
    hostId: 1,
    hostImage: 'https://s3.amazonaws.com/uifaces/faces/twitter/kylefrost/128.jpg',
    hostName: 'Walton',
    hostRes: 'Ipsa rem eius. Eum sed cupiditate earum consequatur sunt non delectus. Voluptas molestias ratione eveniet repudiandae quos eos. Architecto recusandae magnam explicabo aperiam dolores. Et eos et debitis magnam.',
    hostResDate: '2019-12-01T10:45:29.000Z',
    image: 'https://s3.amazonaws.com/uifaces/faces/twitter/artvavs/128.jpg',
    location: 4,
    name: 'Dallas',
    outHos: 1,
    quiRes: 1,
    rating: 5,
    spaCle: 1,
    stySpa: 1,
    userId: 6,
    value: 3,
    _id: 6,
  }],
  staticData: {
    hearts: [38, 39, 33, 38, 37],
    ratingd: ['4.1', '4.2', '4.1', '4.0', '3.9', '4.0'],
  },
};

describe('<Reviews />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Reviews allState={state}/>);
    console.log(wrapper.debug());
  });
  it('should not return an error', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should have a row of days depicting the day of the week', () => {
    expect(wrapper.text()).toContain('Reviews');
  });
  it('should have a row of days depicting the day of the week', () => {
    expect(wrapper.text()).toContain('read more');
  });
});
