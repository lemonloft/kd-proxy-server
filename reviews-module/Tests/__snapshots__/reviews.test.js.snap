// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Reviews /> should not return an error 1`] = `ShallowWrapper {}`;

exports[`Test App component using mount should not return an error 1`] = `ReactWrapper {}`;

exports[`Test App component using shallow should not return an error 1`] = `ShallowWrapper {}`;

exports[`it App component using mount should not return an error 1`] = `ReactWrapper {}`;

exports[`it App component using shallow should not return an error 1`] = `ShallowWrapper {}`;
